electroncash_plugins/fusion/plugin.py

```diff
@@ -660,10 +678,11 @@ def wallet_can_fuse(wallet) -> bool:
         return can_fuse_from(wallet) and can_fuse_to(wallet)
 
     @staticmethod
-    def is_fuz_coin(wallet, coin) -> Optional[bool]:
+    def is_fuz_coin(wallet, coin, fuz_parents = 0) -> Optional[bool]:
         """ Returns True if the coin in question is definitely a CashFusion coin (uses heuristic matching),
         or False if the coin in question is not from a CashFusion tx. Returns None if the tx for the coin
-        is not (yet) known to the wallet (None == inconclusive answer, caller may wish to try again later). """
+        is not (yet) known to the wallet (None == inconclusive answer, caller may wish to try again later).
+        Optionally check recursively if the depth of fused coins is sufficient. """
```

Comment:
Suggest: Somewhere in this function after the cache lookup but before the call to the nested function `check_is_fuz_trx`, maybe doing:

```python
        fuz_parents = min(fuz_parents, 900)  # paranoia: clamp recursion to 900
```
I looked it up and the max recursion depth is 1000. This just makes the code resilient to all pathological mis-use in the future... :P